### PR TITLE
fix erroneous mouse motion when alt-tabbing in CS2

### LINKED ISSUE #11422: Erroneous relative mouse input injected when alt tabbing
Observed in CS2, reproduced in a modified version of WoodenEye 008. 

Not sure what is the root cause of this, but when alt tabbing SDL seems to inject a large sum of relative motion from mouse id 0, resulting in you ending up aiming at the floor or turning horizontally 90 degrees after alt tabbing.

### PR DESCRIPTION
**Pull request by expikr**
closes #11422

@Slartibarty can you verify if this fix works for your game?

### CODE REVIEW

File: src/events/SDL_mouse.c [hunk 1]
```diff
@@ -841,6 +846,11 @@ static void SDL_PrivateSendMouseMotion(Uint64 timestamp, SDL_Window *window, SDL
 
     // Post the event, if desired
     if (SDL_EventEnabled(SDL_EVENT_MOUSE_MOTION)) {
+        if (!relative && window_is_relative) {
+            if (!mouse->relative_mode_warp_motion) return;
```
Comment (slouken) [inline src/events/SDL_mouse.c:850]:
```suggestion
            if (!mouse->relative_mode_warp_motion) {
                return;
            }
```

File: src/events/SDL_mouse.c [hunk 2]
```diff
@@ -814,25 +814,30 @@ static void SDL_PrivateSendMouseMotion(Uint64 timestamp, SDL_Window *window, SDL
...
+            mouse->last_y = y;
         }
-    } else {
-        mouse->x = x;
-        mouse->y = y;
         mouse->has_position = true;
     }
 
-    mouse->xdelta += xrel;
-    mouse->ydelta += yrel;
```
[the remark this answers is not in the document]
Comment (slouken) [reply src/events/SDL_mouse.c:834]: Oh, and renamed, I see.

**Comment by Slartibarty** [comment]
> closes #11422
> 
> @Slartibarty can you verify if this fix works for your game?

Yep, this fixes it in my game.

**Comment by slouken** [comment]
Merged, thanks!

**Pull request merged**